packages/vc-export/src/documentLoader.ts

```diff
@@ -78,17 +81,65 @@ export const kiltContextsLoader: DocumentLoader = async (url) => {
   throw new Error(`not a known Kilt context: ${url}`)
 }
 
+type LegacyVerificationMethodType =
+  | 'Sr25519VerificationKey2020'
+  | 'Ed25519VerificationKey2018'
+  | 'EcdsaSecp256k1VerificationKey2019'
+  | 'X25519KeyAgreementKey2019'
+type LegacyVerificationMethod = Pick<
+  VerificationMethod,
+  'id' | 'controller'
+> & { publicKeyBase58: string; type: LegacyVerificationMethodType }
+
+// Returns legacy representations of a KILT DID verification method.
 export const kiltDidLoader: DocumentLoader = async (url) => {
   const { did } = parse(url as DidUri)
-  const { didDocument, didResolutionMetadata } = await resolveCompliant(did)
-  if (didResolutionMetadata.error) {
-    throw new Error(
-      `${didResolutionMetadata.error}:${didResolutionMetadata.errorMessage}`
-    )
+  const { dereferencingMetadata, contentStream } = await dereferenceDid(did)
```

Comment:
Yup. That's a typo from an old version where I used to resolve it directly in the JSON-LD format, which `resolve` does not support. But since now we always resolve the full DID document in pure JSON, I changed it to use the `resolve` instead: https://github.com/KILTprotocol/sdk-js/pull/799/commits/573373ddd52d93d4d85878d0fd12f770408ffc49.

In general `dereference` should be used when only a verification method/service is being resolved (would make even more sense to do so once and if https://github.com/decentralized-identity/did-spec-extensions/pull/21 gets merged), or when a DID Document in a format other than `application/did+json` is to be returned.